calico_containers/adapter/datastore.py

```diff
@@ -228,17 +249,44 @@ def from_json(cls, ep_id, json_str):
         ipv6_gw = json_dict["ipv6_gateway"]
         if ipv6_gw:
             ep.ipv6_gateway = IPAddress(ipv6_gw)
-        ep.profile_id = json_dict["profile_id"]
+
+        # Version controlled fields
+        profile_id = json_dict.get("profile_id", None)
+        ep.profile_ids = [profile_id] if profile_id else \
+                         json_dict.get("profile_ids", [])
+        ep.if_name = json_dict.get("container:if_name", VETH_NAME)
+
+        # Store the original JSON representation of this Endpoint.
+        ep._original_json = json_str
+
         return ep
 
+    def matches(self, hostname=None, orchestrator_id=None, workload_id=None, endpoint_id=None):
+        """
+        A less strict 'equals' function, which compares designated parameters to the current endpoint object.
```

Comment:
can be more specific than designated parameters.  this function compares the IDs that fully qualify (or uniquely identify) a particular endpoint